src/parser/jobs/vpr/modules/RattlingCoil.tsx

```diff
@@ -0,0 +1,110 @@
+import {t} from '@lingui/macro'
+import {Trans, Plural} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import {JOBS} from 'data/JOBS'
+import {Event, Events} from 'event'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+import {CounterGauge, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+type GaugeModifier = Partial<Record<Event['type'], number>>
+
+const LEFTOVER_COIL_SEVERITY_TIERS = {
+	1: SEVERITY.MINOR,
+	2: SEVERITY.MEDIUM,
+}
+
+const MAX_COILS = 3
+
+export class RattlingCoil extends CoreGauge {
+	static override handle = 'rattlingcoil'
+	static override title = t('vpr.rattlingcoil.title')`Rattling Coil Timeline`
+
+	@dependency private suggestions!: Suggestions
+	@dependency private checklist!: Checklist
+
+	private coilGauge = this.add(new CounterGauge({
+		maximum: MAX_COILS,
+		graph: {
+			handle: 'coils',
+			label: <Trans id="vpr.gauge.resource.coilsLabel">Rattling Coils</Trans>,
+			color: JOBS.VIPER.colour,
+			forceCollapsed: true,
+		},
+	}))
+
+	// Used for Checklist as CounterGauge seems to lack total tracking at this time.
+	private totalGeneratedCoils = 0
+
+	private CoilModifiers = new Map<number, GaugeModifier>([
+
+		[this.data.actions.DREADWINDER.id, {action: 1}],
+		[this.data.actions.PIT_OF_DREAD.id, {action: 1}],
+		[this.data.actions.SERPENTS_IRE.id, {action: 1}],
+
+		[this.data.actions.UNCOILED_FURY.id, {action: -1}],
+	])
+
+	override initialise() {
+		super.initialise()
+
+		const coilActions = Array.from(this.CoilModifiers.keys())
+
+		this.addEventHook(
+			filter<Event>()
+				.source(this.parser.actor.id)
+				.type(oneOf(['action', 'combo']))
+				.action(oneOf(coilActions)),
+			this.onGaugeModifier,
+		)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onGaugeModifier(event: Events['action' | 'combo']) {
+		const modifier = this.CoilModifiers.get(event.action)
+
+		if (modifier != null) {
+			const amount = modifier[event.type] ?? 0
+
+			if (amount > 0) {
+				this.totalGeneratedCoils += amount //Increment total tracker for generated ammo
+			}
+			this.coilGauge.modify(amount)
+		}
+	}
+
+	private onComplete() {
+		this.suggestions.add(new TieredSuggestion({
```

Comment:
This suggestion seems to work in support of helping the player meet the checklist goal. Should we also have a suggestion about the overCap value as well, if they did in fact overcap the gauge at some point? Obviously not worth a "don't die with coils" suggestion since we already suggest not dying!

Reply:
Overcap suggestion has been added.